src/t_zset.c

```diff
@@ -1580,6 +1600,22 @@ void zaddGenericCommand(client *c, int flags) {
             "XX and NX options at the same time are not compatible");
         return;
     }
+    if (gx && nx) {
+        addReplyError(c,
+            "GX and NX options at the same time are not compatible");
+        return;
+    }
+    if (lx && nx) {
+        addReplyError(c,
+            "LX and NX options at the same time are not compatible");
+        return;
+    }
+    if (gx && lx) {
+        addReplyError(c,
+            "GX and LX options at the same time are not compatible");
+        return;
+    }
+    /* Note that XX is compatible with either GX or LX */
```

Comment:
how about converting this into one check with some message saying these 3 options are mutually exclusive.

Reply:
That seems reasonable.

Reply:
This code is now updated.